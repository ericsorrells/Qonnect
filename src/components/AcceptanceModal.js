// ========================================================================================
import React from 'react';
import Modal from 'react-modal';
// ========================================================================================
import ShowEventDisplay from '../components/ShowEventDisplay';
// ========================================================================================

class AcceptanceModal extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      acceptanceNote: ''
    }

    this.onNoteInput = this.onNoteInput.bind(this);
    this.onFormSubmit = this.onFormSubmit.bind(this);
  }

  onNoteInput(e) {
    event.preventDefault();
    this.setState({ acceptanceNote: e.target.value })
  }

  onFormSubmit(event){
    event.preventDefault();
    this.props.userAcceptEvent({
      acceptanceNote: this.state.acceptanceNote
    })
  }

  render(){
    return (
      <Modal
        isOpen={this.props.modalOpen}
        onRequestClose={this.props.closeModal}
        // style={customStyles}
        contentLabel="Example Modal"
      >
        <h2>Pleases Confirm You Wish To Accept This Event:</h2>
        <ShowEventDisplay event={this.props.event} />
        <form onSubmit={this.onFormSubmit}>
          <label>Acceptance Note:</label>
          <textarea
            onChange={this.onNoteInput}
            value={this.state.acceptanceNote}
          />
          <button type='submit'>Yes, I Accept!</button>
        </form>
      </Modal>
    )
  }
}

export default AcceptanceModal;