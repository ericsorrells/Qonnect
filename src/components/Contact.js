// ========================================================================================
import React from 'react';
// ========================================================================================

const Contact = () => {
  return(
    <h1>
      Coming Soon...
    </h1>
  )
}

export default Contact;