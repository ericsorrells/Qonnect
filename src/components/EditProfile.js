// ========================================================================================
import React from 'react';
import PropTypes from 'prop-types';
// ========================================================================================

class EditProfile extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      name:        '',
      birthday:    '',
      location:    '',
      joinedOn:    '',
      description: ''
    }
    
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleNameChange = this.handleNameChange.bind(this);
    this.handleLocationChange = this.handleLocationChange.bind(this);
    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
  }

  handleSubmit(e) {
    e.preventDefault();
    this.props.addProfile({
      name:        this.state.name,
      location:    this.state.location,
      description: this.state.description
    });
    this.setState({ name: '', location: '', description: '' })
  }

  handleNameChange(e) {
    e.preventDefault();
    this.setState({ name: e.target.value });
  }

  handleLocationChange(e) {
    e.preventDefault();
    this.setState({ location: e.target.value });
  }

  handleDescriptionChange(e) {
    e.preventDefault();
    this.setState({ description: e.target.value });
  }

  render() {
    return(
      <div>
        <form onSubmit={this.handleSubmit}>
          <input
            type        = "text"
            placeholder = "Add Your Name"
            value       = {this.state.name}
            onChange    = {this.handleNameChange}
          />
          <br/>
          <input
            type        = "text"
            placeholder = "Add Your Location"
            value       = {this.state.location}
            onChange    = {this.handleLocationChange}
          />
          <br/>
          <textarea
            type        = "text"
            placeholder = "Add Description"
            value       = {this.state.description}
            onChange    = {this.handleDescriptionChange}
          />
          <br/>
          <button>Add Profile</button>
        </form>
      </div>
    )
  }
}

EditProfile.propTypes = {
  profile: PropTypes.object
}

export default EditProfile;