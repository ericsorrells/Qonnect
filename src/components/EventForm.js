// ========================================================================================
import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
// ========================================================================================

class EventForm extends React.Component {
  constructor(props) {
    super(props)

    const { event } = props;
    this.state = {
      eventName:   event ? event.eventName    : '',
      date:        event ? event.date         : '',
      time:        event ? event.time         : '',
      location:    event ? event.location     : '',
      category:    event ? event.category     : '',
      imageUrl:    event ? event.imageUrl     : '',
      description: event ? event.description  : '',
      focused: ''
    }

    this.onFormSubmit        = this.onFormSubmit.bind(this);
    this.onNameChange        = this.onNameChange.bind(this);
    this.onTimeChange        = this.onTimeChange.bind(this);
    this.onLocationChange    = this.onLocationChange.bind(this);
    this.onCategoryChange    = this.onCategoryChange.bind(this);
    this.onImageUrlChange    = this.onImageUrlChange.bind(this); 
    this.onDescriptionChange = this.onDescriptionChange.bind(this);
  }

  onFormSubmit(e) {
    e.preventDefault();
    const { firstName, lastName } = this.props.profile;

    this.props.onSubmit({
      userName:    `${firstName} ${lastName}`,
      uid:         this.props.auth.uid,
      eventName:   this.state.eventName,
      date:        moment(this.state.date).valueOf(),
      time:        this.state.time,
      location:    this.state.location,
      category:    this.state.category,
      imageUrl:    this.state.imageUrl,
      description: this.state.description
    })
  }

  onNameChange(e) {
    this.setState({ eventName: e.target.value })
  }

  onTimeChange(e) {
    console.log('TIME: ', e.target.value);
    this.setState({ time: e.target.value  })
  }

  onLocationChange(e) {
    this.setState({ location: e.target.value })
  }

  onCategoryChange(e){
    this.setState({ category: e.target.value })
  }

  onImageUrlChange(e) {
    this.setState({ imageUrl: e.target.value })
  }

  onDescriptionChange(e) {
    this.setState({ description: e.target.value })
  }

  onFocusChange({ focused }) {
    this.setState(() => ({ calendarFocused: focused }))
  }

  render() {
    return (
      <div className='form'>
        <form onSubmit={this.onFormSubmit} >
          <label> Event Name: </label>
          <input
            type='text'
            onChange={this.onNameChange}
            value={this.state.name}
          />

          <label> Event Date: </label>
          <SingleDatePicker
            date={this.state.date}
            onDateChange={ date => this.setState({ date: date }) }
            focused={this.state.focused}
            onFocusChange={({ focused }) => this.setState({ focused })}
            numberOfMonths={1}
            isOutsideRange={() => false} 
          />

          <label>Time:</label>
          <input type='time' onChange={this.onTimeChange}/>
          <label> Event Location: </label>
          <input
            type='text'
            onChange={this.onLocationChange}
            value={this.state.location}
          />

          <label>Category:</label>          
          <select onChange={this.onCategoryChange} value={this.state.category ? this.state.category : ''}>
            <option disabled selected value=''>Select A Category</option>
            <option value="Sports">Sports</option>
            <option value="Entertainment">Entertainment</option>
            <option value="Food">Food</option>
            <option value="Music">Music</option>
            <option value="Miscellaneous">Miscellaneous</option>
          </select>

          <label>Image URL:</label>
          <input
            type='text'
            onChange={this.onImageUrlChange}
            value={this.state.imageUrl}
          />

          <label> Description: </label>
          <textarea
            onChange={this.onDescriptionChange}
            value={this.state.description}
          />

          <button type='submit'>Submit</button>
        </form>
      </div>
    )
  }
}

export default EventForm;