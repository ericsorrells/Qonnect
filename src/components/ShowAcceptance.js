// ========================================================================================
import React from 'react';
import { Link }  from 'react-router-dom';
import { connect } from 'react-redux';
// ========================================================================================
import { startOwnerSelectsGuest } from '../actions/Acceptances_Actions';
import { isEventOwner, formatText } from '../utils/utils';
import { getCurrentUser } from '../firebase/auth';
// ========================================================================================

const ShowAcceptance = (props) => {
  const onSelect = (eventId, acceptanceId) => {
    // TODO: deselect any selected acceptances (ie, ability to change your mind about an acceptance)
    props.startOwnerSelectsGuest(eventId, acceptanceId)
  }

  const { id, userName, acceptanceNote, userPhotoURL, createAt, selected = null } = props.acceptance;
  const { photoURL } = props.user

  return (
    <div className='show-acc_container'>
      <div>
        <Link to={`/profile/${props.acceptance.userId}`} >
          <img src={userPhotoURL} className='show-acc__imageUrl'/>
        </Link>
      </div>
      <div className='show-acc__main-body'>
        <div>
          <div className='show-acc__name-container'>
            <Link to={`/profile/${props.acceptance.userId}`} className='show-acc__name' >
              { userName }
            </Link>
          </div>
          <div className='show-acc__note'>
            { acceptanceNote }
          </div>
        </div>
      </div>
      <div className='show-acc__selection-icon'>
        {
          isEventOwner(props.userId, props.event.uid) && !selected &&
          <a onClick={() => onSelect(props.eventId, id)}><i className='fas fa-thumbs-up'></i></a>
        }
      </div>
    </div>
  )
}

const mapDispatchToProps = (dispatch) => {
  return { 
    startOwnerSelectsGuest: (eventId, acceptanceId) => dispatch(startOwnerSelectsGuest(eventId, acceptanceId))
  }
}

export default connect(null, mapDispatchToProps)(ShowAcceptance);