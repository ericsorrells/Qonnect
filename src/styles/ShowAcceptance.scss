.show-acc_container {
  border-top: 1px solid grey;
  display: flex;
  padding: 8px;
}

.show-acc__main-body {
  display: flex;
  flex-direction: column;
}

.show-acc__name-container {
  margin-bottom: 10px;
}

.show-acc__note {
  color: $grey-text;
  margin-bottom: 5px;
}

.show-acc__name {
  color: black;
  margin-bottom: 10px;
}

.show-acc__imageUrl {
  border-radius: 50%;
  height: 50px;
  width: 50px;
  margin-right: 20px;
}

.show-acc__selection-icon {
  margin-left: auto;
}

.fa-thumbs-up {
  font-size: 2rem;
  padding: 8px;
  color: green;
  border: 1px solid green;
  border-radius: 50%;
}
