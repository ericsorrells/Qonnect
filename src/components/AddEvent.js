// ========================================================================================
import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
// ========================================================================================
import EventItems from './EventItems';
import EventForm from './EventForm';
import Spinner from './Spinner';
// ========================================================================================
import { getCurrentUser } from '../firebase/auth';
import { startGetProfile } from '../actions/Profile_Actions';
import { startCreateEvent } from '../actions/Events_Actions';
import { getEventUserIDs } from '../utils/utils';
// ========================================================================================

class AddEvent extends React.Component {
  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    this.props.startCreateEvent(event)
  }

  componentDidMount() {
    const { userId, events } = this.props;

    let eventIds = null;
    if (events) {
      eventIds = getEventUserIDs(events);
    }
    const hasEvents          = events.length === 0;
    const areUserEvents      = eventIds.includes(userId);

    if (userId && (hasEvents || !areUserEvents)) {
      this.props.startGetProfile(userId);
    }
  }

  render() {
    const component = this.props.loading.loading
      ? <Spinner />
      : <AddEventComponent {...this.props}
        events={this.props.events}
        history={this.props.history}
        deleteEvent={this.props.deleteEvent}
        onSubmit={this.handleSubmit}
        profile={this.props.profile}
        auth={this.props.auth}
      />
    return (
      <div>
        {component}
      </div>
    )
  }
}

const AddEventComponent = ({ events, history, deleteEvent, onSubmit, profile, auth }) => {
  return (
    <div className='add-event-container padding-left'>
      <aside>
        <EventForm
          event={null}
          onSubmit={onSubmit}
          profile={profile}
          auth={auth}
        />
      </aside>
      <section className='add-event__section'>
        <div className='add-event__inner-container'>
          {
            events &&
            <EventItems
              events={events}
              history={history}
              deleteEvent={deleteEvent}
            />
          }
        </div>
      </section>
    </div>
  )
}

AddEvent.propTypes = {
  events:      PropTypes.array,
  history:     PropTypes.func,
  deleteEvent: PropTypes.func,
  userId:      PropTypes.number,
  profile:     PropTypes.object,
  auth:        PropTypes.object,
  loading:     PropTypes.object
};

export { AddEventComponent };
export default AddEvent;