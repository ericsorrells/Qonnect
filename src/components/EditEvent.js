// ========================================================================================
import React from 'react';
// ========================================================================================
import EventForm from '../components/EventForm';
// ========================================================================================

// TODO: Can this be a stateless component???
class EditEvent extends React.Component {
  constructor(props) {
    super(props)
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(event) {
    const { id, eventName, date, time, location, description } = event;
    // TODO: add support for persisting edited data to Firebase
    this.props.editEvent(id, { ...event })
    this.props.history.push(`/show-event/${id}`); 
  }

  render() {
    return (
      <div>
        <EventForm
          event={this.props.event}
          onSubmit={this.onSubmit}
          profile={this.props.profile}
          auth={this.props.auth}
        />
      </div>
    )
  }
}

export default EditEvent;