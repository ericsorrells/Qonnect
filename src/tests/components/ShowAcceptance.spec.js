
// ========================================================================================
import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import mockStoreSetup from '../../utils/mockStoreHelpers';
import 'jest-enzyme';
// ========================================================================================
import { ShowAcceptance } from '../../components/ShowAcceptance';
// ========================================================================================
import { basicAcceptance } from '../stubs/acceptance';
import { basicEvent } from '../stubs/event';
import { basicProfile } from '../stubs/profile';
// ========================================================================================

describe('<ShowAcceptance/>', () => {
  const startOwnerSelectsGuest = jest.fn();
  const eventId = '-LDTTYL9PSqqroO6JLYh';

  const component = shallow(
    <ShowAcceptance
      acceptance={basicAcceptance}
      eventId={eventId}
      userId={basicEvent.uid}
      event={basicEvent}
      user={basicProfile}
      startOwnerSelectsGuest={startOwnerSelectsGuest}
    />
  )

  describe('rendering', () => {
    it('renders correctly', () => {
      expect(component).toMatchSnapshot();
      expect(component.find('.show-acc__name')).toExist();
      expect(component.find('.show-acc__note')).toHaveText(basicAcceptance.acceptanceNote);
      expect(component.find('.fa-thumbs-up')).toExist();
    });
  });

  it('allows event owner to select a guest', () => {
    component.find('a').simulate('click');
    expect(startOwnerSelectsGuest).toHaveBeenCalled();
  });

  it('prevents non-event owners from selecting a guest', () => {
    const acceptanceAlreadyMade = { ...basicAcceptance, selected: true};
    component.setProps({ acceptance: acceptanceAlreadyMade });
    expect(component.find('a')).not.toExist();
  });
});