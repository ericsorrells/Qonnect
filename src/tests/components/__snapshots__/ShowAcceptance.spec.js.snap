// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowAcceptance/> rendering renders correctly 1`] = `
<div
  className="show-acc_container"
>
  <div>
    <Link
      replace={false}
      to="/profile/NClNDy4YHLMdy7ILKcQGcHpI4OD3"
    >
      <img
        className="show-acc__imageUrl"
        src="https://image.jpg"
      />
    </Link>
  </div>
  <div
    className="show-acc__main-body"
  >
    <div>
      <div
        className="show-acc__name-container"
      >
        <Link
          className="show-acc__name"
          replace={false}
          to="/profile/NClNDy4YHLMdy7ILKcQGcHpI4OD3"
        >
          Rachel Deckard
        </Link>
      </div>
      <div
        className="show-acc__note"
      >
        asdfasdf
      </div>
    </div>
  </div>
  <div
    className="show-acc__selection-icon"
  >
    <a
      onClick={[Function]}
    >
      <i
        className="fas fa-thumbs-up"
      />
    </a>
  </div>
</div>
`;
